server/server.go

```diff
@@ -67,6 +71,14 @@ func NewServer(opts ...Option) (*Server, error) {
 	}
 	s.HTTPServer.Handler = mux
 
+	var handler http.Handler
+	if s.middleware != nil {
+		handler = s.middleware(mux)
+	} else {
+		handler = mux
+	}
+	s.HTTPServer.Handler = handler
```

Comment:
It can probably be simplified:

```
	if s.middleware != nil {
		s.HTTPServer.Handler = s.middleware(s.HTTPServer.Handler)
	}
```

Reply:
As I mentioned in comment above, we should consider defining multiple middleware not only one. 

``` go
    for _, m := range s.middleware {
         s.HTTPServer.Handler = m(s.HTTPServer.Handler)
    }
```